recipes/scip/all/conanfile.py

```diff
@@ -0,0 +1,145 @@
+from conan import ConanFile
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import copy, get
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.scm import Version
+from os.path import join
+
+required_conan_version = ">=1.53.0"
+
+
+class SCIPConan(ConanFile):
+    name = "scip"
+    description = "SCIP mixed integer (nonlinear) programming solver"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://scipopt.org/"
+    topics = ("mip", "solver", "linear", "programming")
+    settings = "os", "arch", "compiler", "build_type"
+    package_type = "library"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_boost": [True, False],
+        "with_gmp": [True, False],
+        "with_tpi": ["none", "omp", "tny"],
+        "with_sym": ["bliss", "none"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_boost": True,
+        "with_gmp": True,
+        "with_tpi": "none",
+        "with_sym": "bliss",
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 14
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "5",
+            "clang": "4",
+            "apple-clang": "7",
+        }
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+        if self.options.shared and self.options.with_sym == "bliss":
+            raise ConanInvalidConfiguration("Bliss is not supported in shared mode.")
+        comp = self.settings.compiler
+        if self.options.with_sym == "bliss" and comp == 'clang' and comp.libcxx and comp.libcxx == 'libc++':
+            raise ConanInvalidConfiguration("Bliss does not support libc++.")
```

Comment:
```suggestion
            raise ConanInvalidConfiguration("Bliss does not support libc++.")
        if self.dependencies["soplex"].options.with_boost and not self.options.with_boost:
            raise ConanInvalidConfiguration("The options 'with_boost' should be aligned with 'soplex:with_boost' too.")
        if self.dependencies["soplex"].options.with_gmp and not self.options.with_gmp:
            raise ConanInvalidConfiguration("The options 'with_gmp' should be aligned with 'soplex:with_gmp' too.")
```

In case you are needing it aligned in `configure()` you should validate it too, because users can override it, even if you enforce by configure.